toc/rfc/rfc-draft-service-binding-rotation.md

```diff
@@ -0,0 +1,120 @@
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (fill in with PR link after you submit it)
+
+## Summary
+
+Cloud Controller does not support creating a second binding between an application and a service instance, which prevents seamless rotation of credential bindings.  As a result, the only current method to rotate service credential bindings is through blue-green deployments, a process that many developers are hesitant to use solely for credential rotation. That is why, this RFC proposes to add support for multiple service credential bindings per service instance and application. To simplify the rotation process the RFC suggests a new CF API endpoint and CF CLI extensions.
+
+## Problem
+
+Cloud Controller does not support creation of a second binding for a service instance and application which would enable seamless binding credential rotation. Attempting to create a second binding using the CF CLI results in the following error:
+
+```
+cf bind-service myApp myService --binding-name second-binding
+Binding service instance myService to app myApp in org myOrg / space mySpace as ...
+App myApp is already bound to service instance myService.
+OK
+```
+This limitation is not due to the CF CLI but rather a restriction in the CC API, which prevents multiple bindings for the same application. Attempting to create the binding directly via the CF API results in a similar error:
+
+```
+cf curl -X POST "/v3/service_credential_bindings" -d '{
+    "name": "test-second-binding",
+    "relationships": {
+        "app": {
+            "data": {
+                "guid": "<myApp-guid>"
+            }
+        },
+        "service_instance": {
+            "data": {
+                "guid": "<mySerivce-guid>"
+            }
+        }
+    },
+    "type": "app"
+}'
+
+{"errors":[{"detail":"The app is already bound to the service instance","title":"CF-UnprocessableEntity","code":10008}]}
+```
+As a result, the only way to rotate service credential bindings currently is by performing blue-green deployments. However, many CF application developers are reluctant to push a new version of their application solely for the purpose of credential rotation.
+
+## Proposal
+
+The CC should allow multiple service credential bindings per service instance and application. Only the newest one should be visible in the application VCAP_SERVICES env var (or file-based service binding information) so that existing applications don’t need to change.
+
+### CF Cloud Controller
+
+#### POST /v3/service_credential_bindings
+
+Shall allow the creation of multiple service credential bindings for the same app and service instance under the following conditions:
+- service credential bindings are of type `app`
+- service instance is a managed service instances
+- service credential binding name is not changed
+
+The number of multiple service credential bindings for the same app and service instance should be limited.
```

Comment:
Why does this need to be limited (outside of any existing limits on number of service bindings for an app)?